parser.cpp

```diff
@@ -0,0 +1,101 @@
+#include "xos.h"
+#include <iostream>
+#include <string>
+
+namespace xos
+{
+  std::unique_ptr<PrototypeAST> Parser::parsePrototype(xos::Result<xos::Token> tokIden)
+  {
+    std::string name = tokIden.get().getVal().sval;
+    xos::Result<xos::Token> tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::lparen)
+    {
+      std::cout << "Error: expected ( in prototype." << std::endl;
+      return nullptr;
+    }
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::lparen)
+    {
+      std::cout << "Error: expected ( in prototype." << std::endl;
+      return nullptr;
+    }
+    //[TODO] parse the input arguments here.
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::rparen)
+    {
+      std::cout << "Error: expected ) in prototype." << std::endl;
+      return nullptr;
+    }
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::arrow)
+    {
+      std::cout << "Error: expected => in prototype." << std::endl;
+      return nullptr;
+    }
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::lparen)
+    {
+      std::cout << "Error: expected ( in prototype." << std::endl;
+      return nullptr;
+    }
+    //[TODO] parse the return type arguments.
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::rparen)
+    {
+      std::cout << "Error: expected ) in prototype." << std::endl;
+      return nullptr;
+    }
+    tok = lexer_.getNextToken();
+    if (tok.get().getKind() != xos::Token::rparen)
+    {
+      std::cout << "Error: expected ) in prototype." << std::endl;
+      return nullptr;
+    }
+    return std::make_unique<PrototypeAST>(name, std::vector<std::string>(), "void"); // [Update as per input args and return args]
+  };
+
+  std::unique_ptr<FunctionAST> Parser::parseFunction(std::unique_ptr<PrototypeAST> proto)
+  {
+    xos::Result<xos::Token> tok = lexer_.getNextToken();
+    //[TODO] check for body function here. WELP. WHAT TO DO HERE?
```

Comment:
Let's use this style of TODO comment from https://google.github.io/styleguide/cppguide.html#TODO_Comments

```
TODO(@dsLeks): Check for body function here. WELP. WHAT TO DO HERE?
```